### ISSUE TITLE
Cannot connect with Phillips Hue app (Hue Essentials works)

### ISSUE THREAD

**Issue by michaelmell** [diyhue/diyHue#552]
**Describe the issue**
* I setup DiyHue with docker with the following command:

```
sudo docker run -d --name diyhue_debug -e "DEBUG=true" --restart=always --network=host -e MAC=dc:a6:32:7d:e2:72 -v /home/openhabian/diyhue_docker_mounts/hue-emulator/export:/opt/hue-emulator/export diyhue/core
```

* I use Deconz running on the same RPi 4 also in docker with a Conbee II dongle only (no Hue Bridge). With this setup I can now control my light with the Hue Essentials app running on Android.

* However, I cannot connect with the original Hue App. When I try to connect with the Hue App (i.e. pushing the green "Search" button), the app searches and then very briefly shows something like "1 found bridge" (it is to quick to actually read well), but then immediately shows:

> No new Hue Bridge found

**Steps you tried**

* I have tried clearing the cache of the Hue App on Android and deleting the cert.pem as suggested [here](https://diyhue.org/faq/).
* I also tried chaning the software version number as also suggested on the FAQ.
* I have tried deleting the config.json and starting over.
* I have tried looking at the certificate as suggested [here](https://diyhue.readthedocs.io/en/latest/AddFuncts/debug.html#https-certificate-check).
This only worked as expected without https, which is reminiscent of the issue described [here](https://diyhue.discourse.group/t/not-detected-in-hue-app-problem-with-certificate/127/9). This is the output:

with https:
```
curl https://127.0.0.1/api/nouser/config -v -k
* Expire in 0 ms for 6 (transfer 0x10e58a0)
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Expire in 200 ms for 4 (transfer 0x10e58a0)
* Connected to 127.0.0.1 (127.0.0.1) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: none
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to 127.0.0.1:443 
* Closing connection 0
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to 127.0.0.1:443
```

with http:
```
curl http://127.0.0.1/api/nouser/config -v -k
* Expire in 0 ms for 6 (transfer 0x6ef8a0)
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Expire in 200 ms for 4 (transfer 0x6ef8a0)
* Connected to 127.0.0.1 (127.0.0.1) port 80 (#0)
> GET /api/nouser/config HTTP/1.1
> Host: 127.0.0.1
> User-Agent: curl/7.64.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Server: nginx 
< Date: Sat, 26 Dec 2020 23:23:22 GMT
< Content-type: application/json
< Content-Length: 227
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET, OPTIONS, POST, PUT, DELETE
< Access-Control-Allow-Headers: X-Requested-With
< Access-Control-Allow-Headers: Content-Type
< 
* Connection #0 to host 127.0.0.1 left intact
{"name":"Philips hue","datastoreversion":70,"swversion":"1939070020","apiversion":"1.35.0","mac":"dc:a6:32:7d:XX:XX","bridgeid":"DCA632FFFE7DE272","factorynew":false,"replacesbridgeid":null,"modelid":"BSB002","starterkitid":""}
```

**How we can help**

Please help me figure out what the issue is with my configuration that is causing this connection issue.

**Logs**

This is the output I obtained by connecting to the docker container with `docker container attach diyhue_debug`. I tried connecting with the Hue App, while obtaining this output (I think I tried to connect twice, but not 100% sure anymore):

```
2020-12-26 23:11:41,082 - root - DEBUG - Sending M-Search response to 192.168.178.1
2020-12-26 23:11:46,082 - root - DEBUG - Sending M-Search response to 192.168.178.1
2020-12-26 23:11:51,785 - root - DEBUG - Sending M-Search response to 192.168.178.1
2020-12-26 23:11:52,645 - root - INFO - {"attr":{"id":"1","lastannounced":null,"lastseen":"2020-12-26T23:11Z","manufacturername":"dresden elektronik","modelid":"ConBee II","name":"Configuration tool 1","swversion":"0x264a0700","type":"Configuration tool","uniqueid":"00:21:2e:ff:ff:XX:XX:XX-01"},"e":"changed","id":"1","r":"lights","t":"event","uniqueid":"00:21:2e:ff:ff:XX:XX:XX-01"}
192.168.178.28 - - [26/Dec/2020 23:11:55] "GET /api/nouser/config HTTP/1.1" 200 -
2020-12-26 23:11:56,185 - root - DEBUG - Sending M-Search response to 192.168.178.1
2020-12-26 23:12:01,786 - root - DEBUG - Sending M-Search response to 192.168.178.1
2020-12-26 23:12:02,154 - urllib3.connectionpool - DEBUG - Starting new HTTPS connection (1): discovery.diyhue.org:443
2020-12-26 23:12:02,379 - urllib3.connectionpool - DEBUG - https://discovery.diyhue.org:443 "POST / HTTP/1.1" 200 6
{"ok"}
2020-12-26 23:12:08,510 - root - INFO - {"attr":{"id":"3","lastseen":"2020-12-26T23:12Z","manufacturername":"OSRAM","modelid":"Plug 01","name":"On/Off plug-in unit 3","swversion":"V1.05.09","type":"On/Off plug-in unit","uniqueid":"7c:b0:3e:aa:0a:XX:XX:XX-03"},"e":"changed","id":"3","r":"lights","t":"event","uniqueid":"7c:b0:3e:aa:0a:XX:XX:XX-03"}
2020-12-26 23:12:30,142 - root - DEBUG - Sending M-Search response to 192.168.178.41
2020-12-26 23:12:31,545 - root - DEBUG - Sending M-Search response to 192.168.178.41
2020-12-26 23:12:32,401 - urllib3.connectionpool - DEBUG - Starting new HTTPS connection (1): discovery.diyhue.org:443
2020-12-26 23:12:32,619 - urllib3.connectionpool - DEBUG - https://discovery.diyhue.org:443 "POST / HTTP/1.1" 200 6
{"ok"}
192.168.178.28 - - [26/Dec/2020 23:12:39] "GET /api/nouser/config HTTP/1.1" 200 -
```

**Docker Info (please complete the following information):**
 - Architecture (Ex: x86, ARM): ARM, Raspberry Pi 4
 - Docker image version (Ex: latest, arm-05f043b, 128): diyhue/core  (see docker command above)
 - OS (Ex: Windows 10, Ubuntu Buster): Raspbian GNU/Linux 10 (buster), Kernel: Linux 5.4.79-v7l+
 - Hardware (Ex: PC, Raspberry Pi 1/2/3/Zero): Raspberry Pi 4 Model B Rev 1.2  (4GB version)

**Checklist**
- [X] I confirm that this is a help request
- [X] I confirm that I have completed this template to the best of my ability

**Additional context**

**Comment by alexyao2015** [comment]
Those symptoms are typically due to a bad certificate or issues related to https. Ensure you have the correct external mac address set and that you can access the web ui over https (will show invalid certificate but make sure you can access it).

**Comment by michaelmell** [comment]
I tried connecting via `https://192.168.178.31/`, which gives me this output in the browser:

```
Secure Connection Failed

An error occurred during a connection to 192.168.178.31. PR_END_OF_FILE_ERROR

    The page you are trying to view cannot be shown because the authenticity of the received data could not be verified.
    Please contact the website owners to inform them of this problem.
```

`http` works.
Regarding the correct external MAC: I just deleted the whole config-folder and started over from scratch. I also went into my router to check that the MAC shown there is the same, that I pass to the docker run command.

Any idea, what else I might try?

**Comment by michaelmell** [comment]
This is the docker run command that I used by the way:

```
sudo docker run -d --name diyhue_debug -e "DEBUG=true" --restart=always --network=host -e MAC=DC:A6:XX:XX:XX:XX -v /home/openhabian/diyhue_docker_mounts/hue-emulator/export:/opt/hue-emulator/export diyhue/core:latest
```

**Comment by michaelmell** [comment]
So I tried regenerating the cert.pem using the method described [here](https://github.com/mariusmotea/diyHue/issues/322):
```
curl "http://mariusmotea.go.ro:9002/gencert?mac=00:11:22:33:44:55"
```

However that did not help either.

**Comment by michaelmell** [comment]
I just checked open ports with `sudo netstat -tulpn | grep LISTEN`. For some reason deconz is listening on 443.

```
tcp        0      0 0.0.0.0:9001            0.0.0.0:*               LISTEN      6686/node           
tcp        0      0 0.0.0.0:139             0.0.0.0:*               LISTEN      6716/smbd           
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      14033/python3       
tcp        0      0 0.0.0.0:8180            0.0.0.0:*               LISTEN      17801/deCONZ        
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      540/sshd            
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      17801/deCONZ        
tcp        0      0 0.0.0.0:445             0.0.0.0:*               LISTEN      6716/smbd           
tcp6       0      0 :::139                  :::*                    LISTEN      6716/smbd           
tcp6       0      0 :::5007                 :::*                    LISTEN      6685/java           
tcp6       0      0 :::8080                 :::*                    LISTEN      6685/java           
tcp6       0      0 :::22                   :::*                    LISTEN      540/sshd            
tcp6       0      0 :::8443                 :::*                    LISTEN      6685/java           
tcp6       0      0 :::445                  :::*                    LISTEN      6716/smbd           
tcp6       0      0 127.0.0.1:40419         :::*                    LISTEN      6685/java           
tcp6       0      0 127.0.0.1:8101          :::*                    LISTEN      6685/java
```

Could this be the reason for the issue? I am surprised, because I remapped the ports of deconz in the docker run command:

```
sudo docker run -d \
   --name=deconz \
   --net=host \
   -e DECONZ_WEB_PORT=8180 \
   -e DECONZ_WS_PORT=8443 \
   --restart=always \
   -v /etc/localtime:/etc/localtime:ro \
   -v ~/.local/share/dresden-elektronik/deCONZ:/root/.local/share/dresden-elektronik/deCONZ \
   --device=/dev/ttyACM0 \
   -e DECONZ_DEVICE=/dev/ttyACM0 \
   marthoc/deconz:stable
```

See [here](https://github.com/marthoc/docker-deconz) for details on the options.
Any help/comments would be greatly appreciated.

**Comment by michaelmell** [comment]
Update:
I have confirmed that the issue is due to the deconz container continuing to use port 443: When I stop the deconz container, I can then connect to diyHue on HTTPS (which previouly did not work) and also the Hue App then finds the the diyHue bridge and is able to connect to it.
I now need to figure out how to fix the port-issue with the deconz container. I have created an issue for this [here](https://github.com/marthoc/docker-deconz/issues/335) in case somebody wants to help out.
I am closing this issue, because the problem is not related to diyHue.
